### PR TITLE
Removed deprecated method

### PR DESCRIPTION
**Pull request by rfazi**
Removed deprecated method getResources().getDrawable() and replaced with ContextCompat.getDrawable().

Removed deprecated method getResources().getColor() and replaced with ContextCompat.getColor().

Many changes.

### CODE REVIEW

File: app/src/main/java/com/simplecity/amp_library/ui/appwidget/BaseWidgetConfigure.java [hunk 1]
```diff
@@ -1,5 +1,18 @@
 package com.simplecity.amp_library.ui.appwidget;
 
+import com.afollestad.aesthetic.Aesthetic;
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/appwidget/BaseWidgetConfigure.java]: Need to revert the changes to import arrangement.

File: app/src/main/java/com/simplecity/amp_library/ui/dialog/ChangelogDialog.java [hunk 2]
```diff
@@ -1,21 +1,22 @@
 package com.simplecity.amp_library.ui.dialog;
 
+import com.afollestad.aesthetic.Aesthetic;
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/dialog/ChangelogDialog.java]: Need to revert changes to import arrangement. (Same for all affected files).

File: app/src/main/java/com/jp/wasabeef/glide/transformations/MaskTransformation.java [hunk 3]
```diff
@@ -16,20 +16,20 @@
  * limitations under the License.
  */
 
+import com.bumptech.glide.Glide;
```
Comment (timusus) [inline app/src/main/java/com/jp/wasabeef/glide/transformations/MaskTransformation.java]: This is a third party library, prefer not to make changes here, as it makes it difficult to merge in future changes to this library.

File: app/src/main/java/com/jp/wasabeef/glide/transformations/internal/Utils.java [hunk 4]
```diff
@@ -1,38 +0,0 @@
-package com.jp.wasabeef.glide.transformations.internal;
```
Comment (timusus) [inline app/src/main/java/com/jp/wasabeef/glide/transformations/internal/Utils.java]: This is a third party library, prefer not to make changes here, as it makes it difficult to merge in future changes to this library.

File: app/src/main/java/com/android/vending/billing/utils/Base64.java [hunk 5]
```diff
@@ -31,6 +31,8 @@
  * @version 1.3
  */
 
+import com.simplecity.amp_library.BuildConfig;
```
Comment (timusus) [inline app/src/main/java/com/android/vending/billing/utils/Base64.java]: This is a third party library, prefer not to make changes here, as it makes it difficult to merge in future changes to this library.

File: app/src/main/java/com/android/vending/billing/utils/IabHelper.java [hunk 6]
```diff
@@ -536,7 +536,7 @@ public Inventory queryInventory(boolean querySkuDetails, List<String> moreSkus)
     /**
```
Comment (timusus) [inline app/src/main/java/com/android/vending/billing/utils/IabHelper.java]: This is a third party library, prefer not to make changes here, as it makes it difficult to merge in future changes to this library.

File: app/src/main/java/com/android/vending/billing/utils/Inventory.java [hunk 7]
```diff
@@ -70,9 +70,7 @@ public void erasePurchase(String sku) {
     /** Returns a list of all owned product IDs of a given type */
```
Comment (timusus) [inline app/src/main/java/com/android/vending/billing/utils/Inventory.java]: This is a third party library, prefer not to make changes here, as it makes it difficult to merge in future changes to this library.

File: app/src/main/java/com/android/vending/billing/utils/Security.java [hunk 8]
```diff
@@ -47,7 +47,7 @@
     /**
```
Comment (timusus) [inline app/src/main/java/com/android/vending/billing/utils/Security.java]: This is a third party library, prefer not to make changes here, as it makes it difficult to merge in future changes to this library.

File: app/src/main/java/com/simplecity/amp_library/glide/utils/BitmapAndSize.java [hunk 9]
```diff
@@ -19,9 +19,7 @@ public boolean equals(Object o) {
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/glide/utils/BitmapAndSize.java]: I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.

File: app/src/main/java/com/simplecity/amp_library/lyrics/QuickLyricUtils.java [hunk 10]
```diff
@@ -49,8 +49,7 @@ static Intent getQuickLyricIntent() {
      * @return true if the Play Store is available, and this QuickLyric can be downloaded.
      */
     static boolean canDownloadQuickLyric() {
-        if (ShuttleUtils.isAmazonBuild()) return false;
-        return getQuickLyricIntent().resolveActivity(ShuttleApplication.getInstance().getPackageManager()) != null;
+        return !ShuttleUtils.isAmazonBuild() && getQuickLyricIntent().resolveActivity(ShuttleApplication.getInstance().getPackageManager()) != null;
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/lyrics/QuickLyricUtils.java]: I'd prefer not to make this change. 'Simplifying' the conditional statement actually makes it harder to read.

File: app/src/main/java/com/simplecity/amp_library/model/Album.java [hunk 11]
```diff
@@ -166,8 +166,7 @@ public boolean equals(Object o) {
 
         Album album = (Album) o;
 
-        if (id != album.id) return false;
-        return name != null ? name.equals(album.name) : album.name == null;
+        return id == album.id && (name != null ? name.equals(album.name) : album.name == null);
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/model/Album.java]: I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.

File: app/src/main/java/com/simplecity/amp_library/model/AlbumArtist.java [hunk 12]
```diff
@@ -114,9 +114,7 @@ public boolean equals(Object o) {
 
         AlbumArtist that = (AlbumArtist) o;
 
-        if (name != null ? !name.equals(that.name) : that.name != null) return false;
-        return albums != null ? albums.equals(that.albums) : that.albums == null;
-
+        return name != null ? name.equals(that.name) : that.name == null && (albums != null ? albums.equals(that.albums) : that.albums == null);
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/model/AlbumArtist.java]: I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.

File: app/src/main/java/com/simplecity/amp_library/model/Artist.java [hunk 13]
```diff
@@ -21,10 +21,7 @@ public boolean equals(Object o) {
 
         Artist artist = (Artist) o;
 
-        if (id != artist.id) return false;
-        if (numAlbums != artist.numAlbums) return false;
-        if (numSongs != artist.numSongs) return false;
-        return name != null ? name.equals(artist.name) : artist.name == null;
+        return id == artist.id && numAlbums == artist.numAlbums && numSongs == artist.numSongs && (name != null ? name.equals(artist.name) : artist.name == null);
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/model/Artist.java]: I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.

File: app/src/main/java/com/simplecity/amp_library/model/ArtworkModel.java [hunk 14]
```diff
@@ -42,9 +42,7 @@ public boolean equals(Object o) {
 
         ArtworkModel that = (ArtworkModel) o;
 
-        if (type != that.type) return false;
-        return file != null ? file.equals(that.file) : that.file == null;
-
+        return type == that.type && (file != null ? file.equals(that.file) : that.file == null);
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/model/ArtworkModel.java]: I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.

File: app/src/main/java/com/simplecity/amp_library/model/BaseFileObject.java [hunk 15]
```diff
@@ -40,11 +40,7 @@ public boolean equals(Object o) {
 
         BaseFileObject that = (BaseFileObject) o;
 
-        if (size != that.size) return false;
-        if (fileType != that.fileType) return false;
-        if (name != null ? !name.equals(that.name) : that.name != null) return false;
-        return path != null ? path.equals(that.path) : that.path == null;
-
+        return size == that.size && fileType == that.fileType && (name != null ? name.equals(that.name) : that.name == null) && (path != null ? path.equals(that.path) : that.path == null);
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/model/BaseFileObject.java]: I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.

File: app/src/main/java/com/simplecity/amp_library/model/Genre.java [hunk 16]
```diff
@@ -57,10 +57,7 @@ public boolean equals(Object o) {
 
         Genre genre = (Genre) o;
 
-        if (id != genre.id) return false;
-        if (numSongs != genre.numSongs) return false;
-        return name != null ? name.equals(genre.name) : genre.name == null;
-
+        return id == genre.id && numSongs == genre.numSongs && (name != null ? name.equals(genre.name) : genre.name == null);
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/model/Genre.java]: I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.

File: app/src/main/java/com/simplecity/amp_library/model/Playlist.java [hunk 17]
```diff
@@ -274,8 +274,7 @@ public boolean equals(Object o) {
 
         Playlist playlist = (Playlist) o;
 
-        if (id != playlist.id) return false;
-        return name != null ? name.equals(playlist.name) : playlist.name == null;
+        return id == playlist.id && (name != null ? name.equals(playlist.name) : playlist.name == null);
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/model/Playlist.java]: I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.

File: app/src/main/java/com/simplecity/amp_library/model/Song.java [hunk 18]
```diff
@@ -1,17 +1,5 @@
 package com.simplecity.amp_library.model;
 
-import android.content.ContentUris;
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/model/Song.java]: That's a lot of changes to remove a single line of code! If we're committing formatting changes, they need to be consistent with the rest of the app. (Need to revert import rearrangement)

File: app/src/main/java/com/simplecity/amp_library/model/SuggestedHeader.java [hunk 19]
```diff
@@ -19,11 +19,9 @@ public boolean equals(Object o) {
 
         SuggestedHeader that = (SuggestedHeader) o;
 
-        if (title != null ? !title.equals(that.title) : that.title != null) return false;
-        if (subtitle != null ? !subtitle.equals(that.subtitle) : that.subtitle != null)
-            return false;
-        return playlist != null ? playlist.equals(that.playlist) : that.playlist == null;
-
+        return (title != null ? title.equals(that.title) : that.title == null) &&
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/model/SuggestedHeader.java]: I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.

File: app/src/main/java/com/simplecity/amp_library/model/WhitelistFolder.java [hunk 20]
```diff
@@ -19,9 +19,7 @@ public boolean equals(Object o) {
 
         WhitelistFolder that = (WhitelistFolder) o;
 
-        if (id != that.id) return false;
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/model/WhitelistFolder.java]: I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.

File: app/src/main/java/com/simplecity/amp_library/playback/MultiPlayer.java [hunk 21]
```diff
@@ -53,7 +55,12 @@ private boolean setDataSourceImpl(final MediaPlayer mediaPlayer, final String pa
             } else {
                 mediaPlayer.setDataSource(path);
             }
-            mediaPlayer.setAudioStreamType(AudioManager.STREAM_MUSIC);
+            if (ShuttleUtils.hasOreo()) {
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/playback/MultiPlayer.java]: Good change, but why not make this under a separate commit? Totally unrelated to the other changes.

File: app/src/main/java/com/simplecity/amp_library/playback/MusicService.java [hunk 22]
```diff
@@ -1528,8 +1535,8 @@ public void open(List<Song> songs, final int position) {
     /**
      * Moves the item at index1 to index2.
      *
-     * @param from
-     * @param to
+     * @param from index1
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/playback/MusicService.java]: I don't understand this change - the param names are `from` and `to`, not `index1` and `index2`

File: app/src/main/java/com/simplecity/amp_library/playback/MusicService.java [hunk 23]
```diff
@@ -1753,7 +1760,17 @@ public void openFile(String path, @Nullable Action completion) {
      */
     public void play() {
 
-        int status = audioManager.requestAudioFocus(audioFocusListener, AudioManager.STREAM_MUSIC, AudioManager.AUDIOFOCUS_GAIN);
+        int status;
+        if (ShuttleUtils.hasOreo()) {
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/playback/MusicService.java]: Good change, should probably be made under a separate commit. It's very hard to review among all these other changes.

File: app/src/main/java/com/simplecity/amp_library/services/EqualizerService.java [hunk 24]
```diff
@@ -246,11 +246,7 @@ public void onDestroy() {
 
         unregisterReceiver(mAudioSessionReceiver);
 
-        for (EffectSet gone : mAudioSessions.values()) {
-            if (gone != null) {
-                gone.release();
-            }
-        }
+        mAudioSessions.values().stream().filter(s -> s != null).forEach(EffectSet::release);
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/services/EqualizerService.java]: Can we give `s` a more appropriate name here? I can't tell what it's supposed to be.

File: app/src/main/java/com/simplecity/amp_library/ui/activities/BaseActivity.java [hunk 25]
```diff
@@ -102,10 +102,7 @@ public void onServiceDisconnected(ComponentName name) {
     @Override
     public boolean onKeyDown(int keyCode, KeyEvent event) {
         //Fix for issue on LG devices
-        if (keyCode == KeyEvent.KEYCODE_MENU && "LGE".equalsIgnoreCase(Build.BRAND)) {
-            return true;
-        }
-        return super.onKeyDown(keyCode, event);
+        return (keyCode == KeyEvent.KEYCODE_MENU && "LGE".equalsIgnoreCase(Build.BRAND)) || super.onKeyDown(keyCode, event);
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/activities/BaseActivity.java]: This is harder to read than the un-simplified statement.

File: app/src/main/java/com/simplecity/amp_library/ui/detail/PlaylistDetailFragment.java [hunk 26]
```diff
@@ -91,10 +91,7 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
                 }) {
             @Override
             public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
-                if (viewHolder.getItemViewType() == target.getItemViewType()) {
-                    return super.onMove(recyclerView, viewHolder, target);
-                }
-                return false;
+                return viewHolder.getItemViewType() == target.getItemViewType() && super.onMove(recyclerView, viewHolder, target);
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/detail/PlaylistDetailFragment.java]: Harder to read than previously.

File: app/src/main/java/com/simplecity/amp_library/ui/detail/PlaylistDetailFragment.java [hunk 27]
```diff
@@ -213,10 +210,7 @@ protected void setupToolbarMenu(Toolbar toolbar) {
 
     @Override
     public boolean onMenuItemClick(MenuItem item) {
-        if (MenuUtils.handleMenuItemClicks(getContext(), item, playlist)) {
-            return true;
-        }
-        return super.onMenuItemClick(item);
+        return MenuUtils.handleMenuItemClicks(getContext(), item, playlist) || super.onMenuItemClick(item);
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/detail/PlaylistDetailFragment.java]: Harder to read than previously.

File: app/src/main/java/com/simplecity/amp_library/ui/dialog/BiographyDialog.java [hunk 28]
```diff
@@ -66,7 +66,8 @@ public void onResponse(Call<LastFmArtist> call, Response<LastFmArtist> response)
                 progressBar.setVisibility(View.GONE);
                 if (response != null && response.isSuccessful()) {
                     if (response.body() != null && response.body().artist != null && response.body().artist.bio != null) {
-                        message.setText(Html.fromHtml(response.body().artist.bio.summary));
+                        if (ShuttleUtils.hasNougat()) message.setText(Html.fromHtml(response.body().artist.bio.summary, Html.FROM_HTML_MODE_COMPACT));
+                        else message.setText(Html.fromHtml(response.body().artist.bio.summary));
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/dialog/BiographyDialog.java]: Probably a good idea to factor out `response.body().artist.bio.summary)`

File: app/src/main/java/com/simplecity/amp_library/ui/fragments/EqualizerFragment.java [hunk 29]
```diff
@@ -622,9 +622,7 @@ private void setupPresets() {
         eqPresetNames = new String[numPresets + 1];
 
         String[] presetNames = prefs.getString("equalizer.preset_names", "").split("\\|");
-        for (short i = 0; i < numPresets; i++) {
-            eqPresetNames[i] = presetNames[i];
-        }
+        System.arraycopy(presetNames, 0, eqPresetNames, 0, numPresets + 1);
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/fragments/EqualizerFragment.java]: I've never trusted `arrayCopy`. That's totally irrational and unjustified - but I'd be more comfortable with this kind of change being made in a separate commit. Makes it a lot easier to revert if something goes wrong.

File: app/src/main/java/com/simplecity/amp_library/ui/fragments/MainController.java [hunk 30]
```diff
@@ -213,11 +213,7 @@ public FragmentInfo getRootViewControllerInfo() {
 
     @Override
     public boolean consumeBackPress() {
-        if (multiSheetView.consumeBackPress()) {
-            return true;
-        }
-
-        return super.consumeBackPress();
+        return multiSheetView.consumeBackPress() || super.consumeBackPress();
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/fragments/MainController.java]: Harder to read than previous.

File: app/src/main/java/com/simplecity/amp_library/utils/ShuttleUtils.java [hunk 31]
```diff
@@ -209,6 +209,12 @@ public static boolean isUpgraded() {
         } catch (Exception ignored) {
         }
 
+        //Another check if something goes wrong with first package check
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/utils/ShuttleUtils.java]: This goes against the following code: 

>If something goes wrong, assume the user has the pro version

File: app/src/main/java/com/simplecity/amp_library/utils/MusicUtils.java [hunk 32]
```diff
@@ -158,10 +158,7 @@ public static String getFilePath() {
      * @return True if we're playing music, false otherwise.
      */
     public static boolean isPlaying() {
-        if (MusicServiceConnectionUtils.serviceBinder != null && MusicServiceConnectionUtils.serviceBinder.getService() != null) {
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/utils/MusicUtils.java]: Harder to read than previously.

File: app/src/main/java/com/simplecity/amp_library/ui/modelviews/SongView.java [hunk 33]
```diff
@@ -301,14 +293,14 @@ public boolean equals(Object o) {
...
 
-        if (editable != songView.editable) return false;
-        if (showAlbumArt != songView.showAlbumArt) return false;
-        if (showPlayCount != songView.showPlayCount) return false;
-        if (showTrackNumber != songView.showTrackNumber) return false;
-        if (showArtistName != songView.showArtistName) return false;
-        if (showAlbumName != songView.showAlbumName) return false;
-        if (isCurrentTrack != songView.isCurrentTrack) return false;
-        return song != null ? song.equals(songView.song) : songView.song == null;
+        return editable == songView.editable &&
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/modelviews/SongView.java]: Harder to read than previously.

File: app/src/main/java/com/simplecity/amp_library/ui/modelviews/SongView.java [hunk 34]
```diff
@@ -282,11 +278,7 @@ public String getSectionName() {
 
     @Override
     public boolean areContentsEqual(Object other) {
-        if (other instanceof SongView) {
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/modelviews/SongView.java]: Harder to read than previously.

File: app/src/main/java/com/simplecity/amp_library/ui/modelviews/FolderView.java [hunk 35]
```diff
@@ -252,11 +252,7 @@ protected String doInBackground(Void... params) {
         protected void onPostExecute(String s) {
             super.onPostExecute(s);
 
-            if (textView != null) {
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/modelviews/FolderView.java]: Harder to read than previously.

File: app/src/main/java/com/simplecity/amp_library/ui/modelviews/FolderView.java [hunk 36]
```diff
@@ -284,8 +280,6 @@ public boolean areContentsEqual(Object other) {
         if (other == null || getClass() != other.getClass()) return false;
         if (!super.areContentsEqual(other)) return false;
 
-        if (!baseFileObject.equals(((FolderView) other).baseFileObject)) return false;
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/modelviews/FolderView.java]: Harder to read than previously.

File: app/src/main/java/com/simplecity/amp_library/ui/modelviews/AlbumView.java [hunk 37]
```diff
@@ -232,10 +228,7 @@ public int hashCode() {
 
     @Override
     public boolean areContentsEqual(Object other) {
-        if (other instanceof AlbumView) {
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/modelviews/AlbumView.java]: Harder to read than previously.

File: app/src/main/java/com/simplecity/amp_library/ui/modelviews/AlbumView.java [hunk 38]
```diff
@@ -218,8 +215,7 @@ public boolean equals(Object o) {
 
         AlbumView albumView = (AlbumView) o;
 
-        if (viewType != albumView.viewType) return false;
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/modelviews/AlbumView.java]: Harder to read than previously.

File: app/src/main/java/com/simplecity/amp_library/ui/modelviews/AlbumView.java [hunk 39]
```diff
@@ -85,10 +85,7 @@ private void onOverflowClick(View v) {
     }
 
     private boolean onAlbumLongclick(int position) {
-        if (listener != null) {
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/modelviews/AlbumView.java]: Harder to read than previously.

File: app/src/main/java/com/simplecity/amp_library/ui/modelviews/AlbumArtistView.java [hunk 40]
```diff
@@ -196,10 +192,7 @@ public int hashCode() {
 
     @Override
     public boolean areContentsEqual(Object other) {
-        if (other instanceof AlbumArtistView) {
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/modelviews/AlbumArtistView.java]: Harder to read than previously.

File: app/src/main/java/com/simplecity/amp_library/ui/modelviews/AlbumArtistView.java [hunk 41]
```diff
@@ -183,8 +180,7 @@ public boolean equals(Object o) {
 
         AlbumArtistView that = (AlbumArtistView) o;
 
-        if (viewType != that.viewType) return false;
```
Comment (timusus) [inline app/src/main/java/com/simplecity/amp_library/ui/modelviews/AlbumArtistView.java]: Harder to read than previously.

**Comment by rfazi** [comment]
@timusus You can remove this pull request because I'm work for open a new pulls.
